src/components/CCPDashboard/PatientInfoTableWithFilters.tsx

```diff
@@ -0,0 +1,428 @@
+        triageFilteredPatients.push(
+          ...patients.filter(
+            (p: Patient) => p.triageLevel === appliedTriageFilters[i].value
+          )
+        );
+      }
+    } else {
+      triageFilteredPatients = patients;
+    }
+
+    if (appliedStatusFilters.length > 0) {
+      for (let i = 0; i < appliedStatusFilters.length; i++) {
+        statusFilteredPatients.push(
+          ...triageFilteredPatients.filter(
+            (p: Patient) => p.status === appliedStatusFilters[i].value
+          )
+        );
+      }
+    } else {
+      statusFilteredPatients = triageFilteredPatients;
+    }
+
+    if (appliedHospitalFilters.length > 0) {
+      for (let i = 0; i < appliedHospitalFilters.length; i++) {
+        hospitalFilteredPatients.push(
+          ...statusFilteredPatients.filter(
+            (p: Patient) =>
+              p.hospitalId?.name === appliedHospitalFilters[i].value
+          )
+        );
+      }
+    } else {
+      hospitalFilteredPatients = statusFilteredPatients;
+    }
+
+    setFilteredPatients(hospitalFilteredPatients);
+  }, [appliedFilters, patients]);
+
+  React.useEffect(() => {
+    filterPatients();
+  }, [filterPatients, appliedFilters]);
+
+  const handleOpenFilters = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const closePopover = () => {
+    setAnchorEl(null);
+  };
+
+  const resetFilters = () => {
+    setSelectedFilters(appliedFilters);
+  };
+
+  const handleCloseFilters = () => {
+    closePopover();
+    resetFilters();
+  };
+
+  const clearFilters = () => {
+    setSelectedFilters(initialFilters);
+  };
+
+  const handleApplyFilters = () => {
+    closePopover();
+    setAppliedFilters(selectedFilters);
+  };
+
+  const removeFilter = (filterKey, category) => {
+    const updatedFilters = {
+      ...appliedFilters,
+      [category]: { ...appliedFilters[category], [filterKey]: false },
+    };
+    setAppliedFilters(updatedFilters);
+    setSelectedFilters(updatedFilters);
+  };
+
+  const handleCheckboxChange = (
+    filterOption: FilterOption,
+    category: FilterCategory,
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setSelectedFilters({
+      ...selectedFilters,
+      [category]: {
+        ...selectedFilters[category],
+        [event?.target.name]: {
+          ...filterOption,
+          selected: event?.target.checked,
+        },
+      },
+    });
+  };
+
+  const filterOption = (
+    filterOption: FilterOption,
+    category: FilterCategory
+  ) => (
+    <Box className={classes.filterOption} key={filterOption.value}>
+      <Checkbox
+        checked={
+          selectedFilters[category][filterOption.label]?.selected || false
+        }
+        onChange={(e) => handleCheckboxChange(filterOption, category, e)}
+        color="secondary"
+        name={filterOption.label}
+      />
+      <Typography variant="body2">{filterOption.label}</Typography>
+    </Box>
+  );
+
+  return (
+    <>
```

Comment:
this pattern is weird to me, is it supposed to wrap a bunch of adjacent tags? is there a reason we don't want to use a div/span instead?

Reply:
The `Box` component here is wrapping the checkbox plus the text so that the checkbox and associated label are both on one line for each of the options in the filters popover. Also, the `Box` component is by default a `div`, it's just mui's component that comes with certain props. "The Box component packages all the style functions that are exposed in @material-ui/system." I've updated how I'm using it, but also I've personally been avoiding plain divs in general and just trying to use mui components.
![image](https://user-images.githubusercontent.com/22457859/88544173-fe0e5000-cfe6-11ea-8584-23a186c90ce3.png)